src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/ClassDataContract.cs

```diff
@@ -439,16 +439,12 @@ static internal bool IsNonAttributedTypeValidForSerialization(Type type)
             }
         }
 
-        private static string[] s_knownSerializableTypeNames = new string[] {
-                "System.Collections.Queue",
-                "System.Collections.Stack",
-                "System.Globalization.CultureInfo",
-                "System.Version",
+        private static string[] s_knownSerializableGenericTypeNames = new string[] {
```

Comment:
Also, why not just have a single Dictionary for both generic and non-generic types...

Then change `IsKnownSerializableType` to something like:

```csharp
        internal static bool IsKnownSerializableType(Type type)
        {
            // Applies to known types that DCS understands how to serialize/deserialize
            //

            TypeInfo typeInfo = type.GetTypeInfo();
            string fullName = typeInfo.IsGenericType && !typeInfo.IsGenericTypeDefinition ?
                typeInfo.GetGenericTypeDefinition().FullName :
                type.FullName;
            
            return s_knownSerializableTypes.ContainsKey(fullName) ||
                Globals.TypeOfException.IsAssignableFrom(type);
        }
```